icu4c/source/common/messagepattern.cpp

```diff
@@ -437,7 +437,7 @@ MessagePattern::parseMessage(int32_t index, int32_t msgStartLength,
     if(U_FAILURE(errorCode)) {
         return 0;
     }
-    if(nestingLevel>Part::MAX_VALUE) {
+    if(nestingLevel>Part::MAX_NESTED_LEVELS) {
```

Comment:
The test `TestICU22798` is failing on Windows. After local repo, I found the value of `nestingLevel` to be zero here.
In the stack trace, this value is coming from [`MessagePattern::parse`](https://github.com/unicode-org/icu/blob/aefbea616201f69cd800744791baa3ebc4e258f1/icu4c/source/common/messagepattern.cpp#L270).

We might need to calculate `nestingLevel` (perhaps in the preParse stage) or here if it's passed as zero.

Reply:
need to change the limit to 0x03ff, 0x0fff is too large for MSVC.